src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -64,34 +64,39 @@ internal DateTime StartTimeCore
         /// <returns>The converted time.</returns>
         internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
         {
-            // Use the uptime and the current time to determine the absolute boot time.
-            DateTime bootTime = DateTime.UtcNow - Uptime;
-
             // And use that to determine the absolute time for timespan.
-            DateTime dt = bootTime + timespanAfterBoot;
+            DateTime dt = BootTime + timespanAfterBoot;
 
             // The return value is expected to be in the local time zone.
             // It is converted here (rather than starting with DateTime.Now) to avoid DST issues.
             return dt.ToLocalTime();
         }
 
-        /// <summary>Gets the elapsed time since the system was booted.</summary>
-        private static TimeSpan Uptime
+        /// <summary>Gets the system boot time.</summary>
+        private static DateTime BootTime
         {
             get
             {
-                // '/proc/uptime' accounts time a device spends in sleep mode.
-                const string UptimeFile = Interop.procfs.ProcUptimeFilePath;
-                string text = File.ReadAllText(UptimeFile);
-
-                double uptimeSeconds = 0;
-                int length = text.IndexOf(' ');
-                if (length != -1)
+                // '/proc/stat -> btime' gets the boot time.
+                // btime is the time of system boot in seconds since the Unix epoch.
+                // It includes suspended time and is updated based on the system time (settimeofday).
+                const string StatFile = Interop.procfs.ProcStatFilePath;
+                string text = File.ReadAllText(StatFile);
```

Comment:
I'm not familiar enough with Span parsing API to suggest improvements here.

Reply:
cc: @stephentoub

Reply:
The parsing after this looks fine.  The really expensive part here is this call to File.ReadAllText.  That can be improved with more code, but as it's mostly the same as what's already there, I wouldn't hold up this PR for it.  I say mostly because the contents of /proc/stat is significantly larger than the contents of /proc/uptime, so it's going to result in a few larger allocations.

Reply:
Q: are there span apis that would work with ReadAllBytes here? Would that save allocations/encoding processing?

Reply:
What we can do here is:
1. Open file as FileStream
2. Allocate Span<char> (in memory or in the stack if small enough) of size == FileStream.Length
3. Read everything into it
4. Do whatever we did before but on this Span
The question is if file size changes in between... Then we can check it again afterwards and repeat
Q. Don't we have already or plan to have "Span<char> TextReader.ReadAllSpan()"?

Reply:
This actually won't work. I did experiment:

Reply:
> What we can do here is

Right, this is along the lines of what I was thinking, though it looks like the files may be larger than we'd like for the stack, in which case we can rent from ArrayPool.

That said, I'm fine with this being done separately, especially considering that the majority of the costs are already there in the existing code.

Reply:
B.T.W. Isn't this a bug that File.ReadAllBytes("/proc/stat") returns empty array? This is because current implementation uses FileStream.Length for it which is reported as 0 for such files:
https://github.com/dotnet/corefx/blob/a8cfc6a498f0455c59b0030f74cba4dabeee86e1/src/System.IO.FileSystem/src/System/IO/File.cs#L320
"stat /proc/stat" also reports 0
Seems like all files in procfs are reported with 0 size

Reply:
Reported an issue: https://github.com/dotnet/corefx/issues/28343